### DISCUSSION TITLE
Does Kong support FIPS?

### DISCUSSION THREAD

**Discussion by morningspace** [Kong/kong#9105] [Help] [answered]
By reading this [blog post](https://konghq.com/blog/kong-gateway-2-7), it looks Kong Gateway started to support FIPS since v2.7. However, this is for Enterprise edition. For Kong Gateway OSS, it seems doesn't. Also, I found someone asked [a similar question](https://github.com/Kong/kong/issues/6355) in 2020.

I'd like to know if Kong will support FIPS in future. If not, any guidance on how to do it? Thanks.

P.S.:

By [searching the Kong repo](https://github.com/Kong/kong/search?q=FIPS), I found in Kong it does includes some code changes related to FIPS compliance, e.g.: https://github.com/Kong/kong/pull/8095, this PR submitted in 2021.

**Comment by hanshuebner** [comment]
Hello @morningspace,

thank you for your question.  At this point, we have no plans to get Kong Gateway OSS as whole certified as FIPS compliant.  The switch to the BoringSSL library, however, will be made with the upcoming 3.0 release for Kong Gateway OSS, too.

Kind regards,
Hans

**Reply by morningspace** [reply]
Thanks @hanshuebner , so IIUC, Kong Gateway OSS will just partially support FIPS which needs to be verified, right? What's the benefit of switching to BoringSSL library in terms of FIPS? Or there must be some other reason? BTW: When will 3.0 release be GA? I see alpha.1 was coming up just a few days ago.

**Reply by hanshuebner** [reply]
BoringSSL is ["partially FIPS compliant"](https://boringssl.googlesource.com/boringssl/+/master/crypto/fipsmodule/FIPS.md), which is why we're pulling it into Kong.

We're working hard on getting Kong 3.0 out of the door, but I cannot provide an exact date as to when it will be available.

**Comment by hanshuebner** [comment, marked as answer]
@morningspace I've just been notified that BoringSSL (and hence FIPS support) is a feature of Kong Enterprise and is not planned to be included in Kong OSS at this time.  Sorry for the misleading response.

**Comment by lihongbj** [comment]
@hanshuebner, for FIPS feature in Kong Enterprise, how to verify if it is ready?

**Reply by hanshuebner** [reply]
The [Kong Gateway release notes](https://docs.konghq.com/gateway/changelog/) contain information about new features that are available in a release.  FIPS compliance did not make it into 3.0.0, but is planned to be contained in a minor update that will come out in the next few weeks.
